File: sast-go/cases/accuracy/context_sensitive/argument_return_value_passing/argument_passing_reference_003_F/argument_passing_reference_003_F.go

// evaluation information start
// real case = false
// evaluation item = 准确度->上下文敏感分析->参数/返回值传递
// scene introduction = 参数值传递->引用传递->数组
// level = 2
// bind_url = accuracy/context_sensitive/argument_return_value_passing/argument_passing_reference_003_F/argument_passing_reference_003_F
// evaluation information end

package main
import "os/exec"

func argument_passing_reference_003_F(__taint_src interface{}) {
	arr := []interface{}{__taint_src}
	process(arr)
	__taint_sink(arr)
}

func process(inputArr []interface{}) {
	inputArr[0] = "changed"
}

func __taint_sink(o interface{}) {
	_ = exec.Command("sh", "-c", o.(string)).Run()
	}

func main() {
    __taint_src := "taint_src_value"
    argument_passing_reference_003_F(__taint_src)
}

File: sast-python2/case/completeness/single_app_tracing/function_call/argument_passing/argument_passing_reference_003_F.py
# -*- coding: utf-8 -*-
# evaluation information start
# real case = false
# evaluation item = 完整度->单应用跟踪完整度->函数和方法调用->参数传递
# scene introduction = 引用传递数组
# level = 2
# bind_url = completeness/single_app_tracing/function_call/argument_passing/argument_passing_reference_003_F
# evaluation information end
import os


def argument_passing_reference_003_F(taint_src):
    arr = [u'_']
    process(arr)  # 现在可以正确调用process函数
    taint_sink(arr)


def process(input_arr):
    input_arr[0] = u'_'


def taint_sink(o):
    os.system(unicode(o))


if __name__ == u'__main__':
    taint_src = u"taint_src_value"
    argument_passing_reference_003_F(taint_src)

